src/lib/controller/store.test.ts

```diff
@@ -31,25 +30,32 @@ describe("StoreController", () => {
     completions: 1,
   };
 
+  const createMockImplementation = () => {
+    const mockPeprStore = new Store();
+    return {
+      Patch: jest.fn().mockReturnValue(Promise.resolve(mockPeprStore)),
+      InNamespace: jest.fn().mockReturnValue({
+        Get: jest.fn().mockReturnValue(Promise.resolve(mockPeprStore)),
+      }),
+      Watch: jest.fn().mockReturnValue({
+        start: jest.fn().mockReturnValue(Promise.resolve()),
+      }),
+      Apply: jest.fn().mockReturnValue(Promise.resolve(mockPeprStore)),
+      Logs: jest.fn().mockReturnValue(Promise.resolve([] as string[])),
+      Get: jest.fn().mockReturnValue(Promise.resolve(mockPeprStore)),
+      Delete: jest.fn().mockReturnValue(Promise.resolve()),
+    };
+  };
+
   afterEach(() => {
     jest.resetAllMocks();
     jest.useRealTimers();
   });
 
   beforeEach(() => {
     testCapability = new Capability(capabilityConfig);
-    const mockPeprStore = new Store();
-    const defaultMockImplementations = <T extends GenericClass, K extends KubernetesObject>() =>
-      ({
-        Patch: jest.fn().mockResolvedValueOnce(undefined as never),
-        InNamespace: jest.fn().mockReturnValueOnce({
-          Get: jest.fn().mockResolvedValueOnce(mockPeprStore as never),
-        }),
-        Watch: jest.fn().mockReturnValueOnce(undefined),
-        Apply: jest.fn().mockResolvedValueOnce(mockPeprStore as never),
-      }) as unknown as K8sInit<T, K>;
-
-    mockK8s.mockImplementationOnce(defaultMockImplementations);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
We generally prefer to avoid the use of `any`, but in this case properly mocking this object requires a lot of scaffolding that is not worth the effort for the level of confidence these tests provide.